guide/expressions.md

```diff
@@ -262,6 +377,78 @@ let cstr = "Hi\0" as *const str as *const [u8] as *const std::os::raw::c_char;
 ```
 
 
+### Chains of fields and method calls
+
+A chain is a sequence of field accesses and/or method calls. A chain may also
+include the try operator. E.g., `a.b.c().d` or `foo?.bar().baz?`.
+
+Prefer formatting on one line if possible, and the chain is *small*. If
+formatting on multiple lines, each field access or method call in the chain
+should be on it's own line with the line-break before the `.` and after any `?`.
+Each line should be block-indented. E.g.,
+
+```rust
+let foo = bar
+    .baz?
+    .qux();
+```
+
+If the length of the last line of the first element plus it's indentation is
+less than or equal to the indentation of the second line (and there is space),
+then combine the first and second lines, e.g.,
+
+```rust
+x.baz?
+    .qux()
+
+let foo = x
+    .baz?
+    .qux();
+
+foo(
+    expr1,
+    expr2,
+).baz?
+    .qux();
+```
+
+#### Multi-line elements
+
+If any element in a chain is formatted across multiple lines, then that element
```

Comment:
IIRC shouldn't we force multi-line style for all elements if any element except the last is multi-lined? Or does this mean that the tool should be configurable?

Reply:
Shouldn't be configurable (necessarily), I think this is a slightly more flexible version of "except the last", so it allows things like:

```
foo.bar
    .baz {
        ...
    }
    .qux;
```